### PR TITLE
UAF-69 / UAF-3601 1099 Reporting Step 10 (1099 Exception Report)

### PR DESCRIPTION
**Pull request by adamkost**
This contains the Form, Action, and BOs necessary to create a 1099 Exception Report.

Note: KFS-CORE and KFS-PURAP both contain ExceptionReport.properties that is an exact copy of KFS-BC's BudgetOrgReport.properties, which is what was referenced in the original KFS3 code. Because KFS-BC is not accessible from KFS-PURAP or KFS-CORE, this file was copied and renamed to fit this file's current use. Why that file was used? I don't know. The file needs to be located in both kfs-core and kfs-purap until the KFS-TAX refactor is completed. Both copies are needed for it to work properly.

### CODE REVIEW

File: kfs-purap/src/main/resources/edu/arizona/kfs/module/purap/spring-purap.xml [hunk 1]
```diff
@@ -220,4 +220,6 @@
 		</property>
     </bean>
 
+	<bean id="taxElectronicFilingService" class="edu.arizona.kfs.module.purap.document.service.impl.ElectronicFilingServiceImpl" />
```
Comment (ghost) [inline kfs-purap/src/main/resources/edu/arizona/kfs/module/purap/spring-purap.xml]: Should inject service properties businessObjectService, parameterService, etc.

File: kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/service/impl/ElectronicFilingServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+	private static final String NINE_SPACES = "         ";
+
+	public BusinessObjectService getBusinessObjectService() {
+		if (businessObjectService == null) {
+			businessObjectService = SpringContext.getBean(BusinessObjectService.class);
+		}
+		return businessObjectService;
+	}
+
```
Comment (ghost) [inline kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/service/impl/ElectronicFilingServiceImpl.java]: Need setter methods for injection of service properties businessObjectService, parameterService, etc.

File: kfs-core/src/main/java/edu/arizona/kfs/tax/businessobject/TransmitterRecord.java [hunk 3]
```diff
@@ -0,0 +1,513 @@
...
+			errors.add(new ElectronicFileException(transmitterTin, companyName1, ElectronicFileException.TRANSMITTER, "Company City ", companyCity, "Company city is a required field. [limit = 40]"));
+		}
+		if (!RecordUtil.isValidStateCode(companyState)) {
+			errors.add(new ElectronicFileException(transmitterTin, companyName1, ElectronicFileException.TRANSMITTER, "Company State", companyState, "Company state is not a valid code. [limit = 2]"));
+		}
+		if (!RecordUtil.isValidZipCode(companyZipCode)) {
+			errors.add(new ElectronicFileException(transmitterTin, companyName1, ElectronicFileException.TRANSMITTER, "Company Zip Code", companyZipCode, "Company zip code is not a valid code. [limit = 9]"));
+		}
+		if ("1".equals(foreignEntityIndicator) && !RecordUtil.isValidCountryCode(companyCountryCode)) {
+			errors.add(new ElectronicFileException(transmitterTin, companyName1, ElectronicFileException.TRANSMITTER, "Company Country Code", foreignEntityIndicator, "Company country code is not a valid code. [limit = 2]"));
```
Comment (ghost) [inline kfs-core/src/main/java/edu/arizona/kfs/tax/businessobject/TransmitterRecord.java]: Should be: 
`errors.add(new ElectronicFileException(transmitterTin, companyName1, ElectronicFileException.TRANSMITTER, "Company Country Code", companyCountryCode, "Company country code is not a valid code. [limit = 2]"));`

**Review by ghost — COMMENTED** [review]
Initial review complete. Please see my comments.

File: kfs-core/src/main/java/edu/arizona/kfs/tax/businessobject/TransmitterRecord.java (hunk 3 above)
Comment (adamkost) [reply kfs-core/src/main/java/edu/arizona/kfs/tax/businessobject/TransmitterRecord.java]: Changed. This appears to have been this way in the original code, so generated reports may differ slightly.

File: kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/service/impl/ElectronicFilingServiceImpl.java (hunk 2 above)
Comment (adamkost) [reply kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/service/impl/ElectronicFilingServiceImpl.java]: Fixed. In making the change to Spring Injection, it seems that 2 helper objects are unused. They have been removed.

File: kfs-purap/src/main/resources/edu/arizona/kfs/module/purap/spring-purap.xml (hunk 1 above)
Comment (adamkost) [reply kfs-purap/src/main/resources/edu/arizona/kfs/module/purap/spring-purap.xml]: Fixed. In making the change to Spring Injection, it seems that 2 helper objects are unused. They have been removed.

**Comment by ghost** [comment]
Looks good!

**Pull request merged**